zilliqa/src/node.rs

```diff
@@ -433,20 +433,50 @@ impl Node {
             return Ok(());
         };
 
+        // Need to get and send corresponding TXs for the block
+        let txs: Vec<SignedTransaction> = block
+            .transactions
+            .iter()
+            .map(|tx_hash| {
+                self.consensus
+                    .get_transaction_by_hash(*tx_hash)
+                    .unwrap()
+                    .unwrap()
+            })
+            .collect::<Vec<_>>();
+
         self.message_sender.send_external_message(
             source,
-            ExternalMessage::BlockResponse(BlockResponse { block }),
+            ExternalMessage::BlockResponse(BlockResponse {
```

Comment:
the block sync mechanism now needs to also include TXs since you need to execute the block. So it now returns proposals. I chose this because it was cleaner/faster, but the most optimal would be to rely on the TX sync mechanism being so good you never don't have one.